spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -878,9 +895,12 @@ private static void rename(File source, File target) throws IOException {
 
 		private volatile long lastWrite;
 
+		private volatile long firstWrite;
```

Comment:
Since this property is set only from ctor and is never update, I guess it should be `final`.
And from here we realize that `FileState(BufferedOutputStream)` ctor should be supplied with population of this property as well.

No? Am I missing anything else?